internal/factsengine/gatherers/ini_files.go

```diff
@@ -0,0 +1,164 @@
+package gatherers
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/trento-project/agent/internal/core/sapsystem"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"gopkg.in/ini.v1"
+)
+
+const (
+	IniFilesGathererName = "ini_files"
+)
+
+// nolint:gochecknoglobals
+var (
+	IniFilesError = entities.FactGatheringError{
+		Type:    "ini-files-error",
+		Message: "ini file error",
+	}
+
+	IniFilesNotFoundError = entities.FactGatheringError{
+		Type:    "ini-files-not-found-error",
+		Message: "cannot find ini file",
+	}
+
+	IniFilesParseError = entities.FactGatheringError{
+		Type:    "ini-files-parse-error",
+		Message: "cannot parse ini file",
+	}
+
+	IniFilesFormatError = entities.FactGatheringError{
+		Type:    "ini-files-format-error",
+		Message: "cannot format ini file content to fact value",
+	}
+)
+
+type IniFilesGatherer struct {
+	fs afero.Fs
+}
+
+func NewIniFilesGatherer(fs afero.Fs) *IniFilesGatherer {
+	return &IniFilesGatherer{fs: fs}
+}
+
+func NewDefaultIniFilesGatherer() *IniFilesGatherer {
+	return &IniFilesGatherer{fs: afero.NewOsFs()}
+}
+
+func (g *IniFilesGatherer) Gather(ctx context.Context, factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	log.Infof("Starting %s facts gathering process", IniFilesGathererName)
+	facts := []entities.Fact{}
+
+	for _, factReq := range factsRequests {
+
+		switch factReq.Argument {
+		case "global.ini":
+			fact, err := g.gatherGlobalIni(ctx, factReq)
+			if err != nil {
+				return nil, errors.Wrap(err, "error gathering global.ini")
+			}
+			facts = append(facts, fact)
+		default:
+			return nil, fmt.Errorf("unsupported ini file for request %s, file: %s", factReq.Name, factReq.Argument)
+		}
+
+	}
+
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
+	return facts, nil
+}
+
+func (g *IniFilesGatherer) gatherGlobalIni(_ context.Context, factRequest entities.FactRequest) (entities.Fact, error) {
+	sids, err := findSIDs(g.fs)
+	if err != nil {
+		return entities.Fact{}, err
+	}
+	if len(sids) == 0 {
+		return entities.Fact{}, errors.New("no SAP system found")
+	}
+
+	values := entities.FactValueList{}
+
+	for _, sid := range sids {
+		path := globalIniPath(sid)
```

Comment:
nitpick for the future: No need to refactor now, but I guess we will need to somehow duplicate this if we introduce new files to parse.
Maybe passing the function to get the path as param could work.
Just to not duplicate the whole function